mlflow/gateway/providers/utils.py

```diff
@@ -22,7 +22,17 @@ async def send_request(headers: Dict[str, str], base_url: str, path: str, payloa
         url = append_to_uri_path(base_url, path)
         timeout = aiohttp.ClientTimeout(total=MLFLOW_GATEWAY_ROUTE_TIMEOUT_SECONDS)
         async with session.post(url, json=payload, timeout=timeout) as response:
-            js = await response.json()
+            content_type = response.headers.get("Content-Type")
+            if content_type and "application/json" in content_type:
+                js = await response.json()
+            elif content_type and "text/plain" in content_type:
```

Comment:
Invalid POST content to MosaicML does not return `application/json` format; instead it returns `text/plain` for reporting the underlying issue.